stacks/ecs.yml

```diff
@@ -462,6 +468,104 @@ Resources:
         - !Ref VPCSubnet1
         - !Ref VPCSubnet2
         - !Ref VPCSubnet3
+  # ECS task draining
+  ECSDrainSNSTopic:
+    Type: "AWS::SNS::Topic"
+    DependsOn: ECSDrainFunction
+    Properties:
+      Subscription:
+        - Endpoint: !GetAtt ECSDrainFunction.Arn
+          Protocol: "lambda"
+  ECSDrainSNSRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: "2012-10-17"
+        Statement:
+          - Effect: "Allow"
+            Principal:
+              Service:
+                - "autoscaling.amazonaws.com"
+            Action:
+              - "sts:AssumeRole"
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/service-role/AutoScalingNotificationAccessRole
+      Path: "/"
+  ECSDrainLambdaRole:
+    Type: "AWS::IAM::Role"
+    Properties:
+      Policies:
+        -
+          PolicyName: "drain-lambda-inline"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
```

Comment:
I wouldn't hate if this gotten broken up a bit with more granular resource allowances, at least for the actions that aren't GET-like (:submit*, :update*, and :publish*, I guess)